metrics/open-telemetry/src/main/scala/sttp/client3/opentelemetry/OpenTelemetryMetricsBackend.scala

```diff
@@ -0,0 +1,175 @@
+          .record((request.contentLength: Option[Long]).map(_.toDouble).getOrElse(0), config.attributes)
+      )
+  }
+
+  private def after[R >: PE, T](request: Request[T, R], e: Throwable): Unit = {
+    incrementCounter(requestToFailureCounterMapper(request, e))
+    decrementUpDownCounter(request)
+  }
+
+  private def decrementUpDownCounter[R >: PE, T](request: Request[T, R]): Unit = {
+    requestToInProgressCounterNameMapper(request)
+      .foreach(config => getOrCreateMetric(upAndDownCounter, config, createNewUpDownCounter).add(-1, config.attributes))
+  }
+
+  private def incrementCounter(collectorConfig: Option[CollectorConfig]): Unit = {
+    collectorConfig
+      .foreach(config => getOrCreateMetric(counters, config, createNewCounter).add(1, config.attributes))
+  }
+
+  override def close(): F[Unit] = delegate.close()
+
+  override def responseMonad: MonadError[F] = delegate.responseMonad
+
+  private def getOrCreateMetric[T](
+      cache: ConcurrentHashMap[String, T],
+      data: CollectorConfig,
+      create: CollectorConfig => T
+  ): T =
+    cache.computeIfAbsent(
+      data.name,
+      new java.util.function.Function[String, T] {
+        override def apply(t: String): T = create(data)
+      }
+    )
+
+  private def createNewUpDownCounter(collectorConfig: CollectorConfig): LongUpDownCounter =
+    meter
+      .upDownCounterBuilder(collectorConfig.name)
+      .setUnit(collectorConfig.unit)
+      .setDescription(collectorConfig.description)
+      .build()
+
+  private def createNewCounter(collectorConfig: CollectorConfig): LongCounter =
+    meter
+      .counterBuilder(collectorConfig.name)
+      .setUnit(collectorConfig.unit)
+      .setDescription(collectorConfig.description)
+      .build()
+
+  private def createNewHistogram(collectorConfig: CollectorConfig): DoubleHistogram =
+    meter
+      .histogramBuilder(collectorConfig.name)
+      .setUnit(collectorConfig.unit)
+      .setDescription(collectorConfig.description)
+      .build()
+}
+
+case class CollectorConfig(
+    name: String,
+    description: String = "",
+    unit: String = "",
+    attributes: Attributes = Attributes.empty()
+)
+case class MeterConfig(name: String, version: String)
+
+object OpenTelemetryMetricsBackend {
+
+  val DefaultRequestsInProgressCounterName = "sttp3_requests_in_progress"
+  val DefaultSuccessCounterName = "sttp3_requests_success_count"
+  val DefaultErrorCounterName = "sttp3_requests_error_count"
+  val DefaultFailureCounterName = "sttp3_requests_failure_count"
+  val DefaultRequestHistogramName = "sttp3_request_size_bytes"
+  val DefaultResponseHistogramName = "sttp3_response_size_bytes"
+
+  def apply[F[_], P](
+      delegate: SttpBackend[F, P],
+      openTelemetry: OpenTelemetry,
+      meterConfig: Option[MeterConfig] = None,
+      requestToInProgressCounterNameMapper: Request[_, _] => Option[CollectorConfig] = (_: Request[_, _]) =>
+        Some(CollectorConfig(DefaultRequestsInProgressCounterName)),
```

Comment:
shouldn't we provide some units as default?

Reply:
In fact unit itself is optional and we are using it in many contexts so I am not sure what exactly should be used as default unit